core/src/main/scala/algebra/lattice/Logic.scala

```diff
@@ -0,0 +1,79 @@
+package algebra
+package lattice
+
+import scala.{specialized => sp}
+
+/**
+  * Logic models a logic generally. It is a bounded distributive
+  * lattice with an extra negation operator.
+  *
+  * The negation operator obeys the weak De Morgan laws:
+  *  - ¬(x∨y) = ¬x∧¬y
+  *  - ¬(x∧y) = ¬¬(¬x∨¬y)
+  *
+  * For intuitionistic logic see [[Heyting]]
+  * For fuzzy logic see [[DeMorgan]]
+  */
+trait Logic[@sp(Int, Long) A] extends Any with BoundedDistributiveLattice[A] { self =>
+  def and(a: A, b: A): A
+
+  def or(a: A, b: A): A
+
+  def not(a: A): A
+
+  def xor(a: A, b: A): A = or(and(a, not(b)), and(not(a), b))
+  def nand(a: A, b: A): A = not(and(a, b))
+  def nor(a: A, b: A): A = not(or(a, b))
+  def nxor(a: A, b: A): A = not(xor(a, b))
+}
+
+trait LogicGenBoolOverlap[H[A] <: Logic[A]] {
+  def and[@sp(Int, Long) A](x: A, y: A)(implicit ev: H[A]): A =
```

Comment:
What is the point of this trait, if it is only extended once (by `object Logic`, but not by `object GenBool`? Same goes for `DeMorganGenboolOverlap`.